src/core/thread/mesh_forwarder_ftd.cpp

```diff
@@ -695,6 +696,22 @@ void MeshForwarder::HandleSentFrameToChild(const Mac::Frame &aFrame, otError aEr
             // address mode for source address matching.
 
             mSourceMatchController.SetSrcMatchAsShort(*child, true);
+
+#if !OPENTHREAD_CONFIG_DROP_MESSAGE_ON_FRAGMENT_TX_FAILURE
+
+            // When `CONFIG_DROP_MESSAGE_ON_FRAGMENT_TX_FAILURE` is
+            // disabled, all fragment frames of a larger message are
+            // sent even if the transmission of an earlier fragment fail.
+            // Note that `GetIndirectTxSuccess() tracks the tx success of
+            // the entire message to the child, while `txError = aError`
+            // represents the error status of the last fragment frame
+            // transmission.
+
+            if (child->GetIndirectTxSuccess() && (txError == OT_ERROR_NONE))
```

Comment:
Should this be `if (!child->GetIndirectTxSuccess() && (txError == OT_ERROR_NONE))`, note the `!` in front.

Reply:
Yes. Thanks. It's correct in the other part.